domain/src/Aggregates/SharePoolingAsset/Entities/SharePoolingAssetTransaction.php

```diff
@@ -15,48 +15,16 @@ abstract class SharePoolingAssetTransaction implements Stringable
 {
     use HasFactory;
 
-    protected bool $processed;
-    protected ?int $position = null;
+    public readonly SharePoolingAssetTransactionId $id;
 
     public function __construct(
         public readonly LocalDate $date,
         public readonly Quantity $quantity,
         public readonly FiatAmount $costBasis,
+        public readonly bool $processed = true,
+        ?SharePoolingAssetTransactionId $id = null,
     ) {
-    }
-
-    abstract public function copy(): static;
-
-    /** @throws SharePoolingAssetTransactionException */
-    public function setPosition(?int $position = null): static
-    {
-        if (! is_null($this->position)) {
-            throw SharePoolingAssetTransactionException::positionAlreadySet($this, $this->position);
-        }
-
-        $this->position = $position;
-
-        return $this;
-    }
-
-    public function getPosition(): ?int
-    {
-        return $this->position;
-    }
-
-    public function isProcessed(): bool
-    {
-        return $this instanceof SharePoolingAssetAcquisition ? true : $this->processed;
-    }
-
-    public function averageCostBasisPerUnit(): ?FiatAmount
-    {
-        return $this->costBasis->dividedBy($this->quantity);
-    }
-
-    public function hasSameDayQuantity(): bool
-    {
-        return $this->sameDayQuantity()->isGreaterThan('0');
+        $this->id = $id ?? SharePoolingAssetTransactionId::generate();
```

Comment:
ℹ️ This is where the object generates an ID for itself (unless one has been provided already)